Description: Auto retry with special options after receiving know error codes from OSX.
 .
 cifs-utils (2:6.1-0.netgear1) unstable; urgency=low
 .
   * Non-maintainer upload.
   * Auto retry with special options after receiving know error codes from OSX.
Author: Justin Maggard <justin.maggard@netgear.com>

---

--- cifs-utils-6.1.orig/mount.cifs.c
+++ cifs-utils-6.1/mount.cifs.c
@@ -2064,6 +2064,20 @@ mount_retry:
 			goto do_mtab;
 
 		switch (errno) {
+		case EOPNOTSUPP:
+			if (strstr(options, "nounix") == NULL) {
+				strlcat(parsed_info->options, ",nounix",
+					sizeof(parsed_info->options));
+				goto mount_retry;
+			}
+			break;
+		case EINVAL:
+			if (strstr(options, "sec=") == NULL) {
+				strlcat(parsed_info->options, ",sec=ntlmssp",
+					sizeof(parsed_info->options));
+				goto mount_retry;
+			}
+			break;
 		case ECONNREFUSED:
 		case EHOSTUNREACH:
 			currentaddress = nextaddress;
